src/Core/Lock/FlockLock.php

```diff
@@ -62,18 +62,26 @@ public function __construct($fileName)
     /**
      * Acquires a lock that will block until released.
      *
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type bool $blocking Whether the process should block while waiting
+     *           to acquire the lock. **Defaults to** true.
+     *     @type bool $exclusive If true, acquire an excluse (write) lock. If
```

Comment:
Since the exclusive flag is used only in the flock case, what are your thoughts on adding it to the constructor for the implementation instead? The big downside I see there is that you would need to instantiate two implementations in `FileBreakpointStorage` - but on the other end future lock implementations won't need to worry about special code to handle the case they may not support the flag.

Reply:
It would seem weird to move one of the options the constructor, but not the other. 

Also, I do think it'd be weird to separate the logic of the type of lock from the actual `acquire()` call. If we are allowing dependency injection for providing a lock implementation to a class that uses a lock, you'd have to know what kind of lock options are necessary for that lock's usage at creation time.

Reply:
> It would seem weird to move one of the options the constructor, but not the other.

I agree it is weird, but it seems equally weird to have an argument in the interface signature that isn't used by most implementations.

> Also, I do think it'd be weird to separate the logic of the type of lock from the actual acquire() call. If we are allowing dependency injection for providing a lock implementation to a class that uses a lock, you'd have to know what kind of lock options are necessary for that lock's usage at creation time.

Very fair point. What about having another implementation specifically for shared locks, and use the current one for exclusive locks? Just thinking out loud. Not sure what is the absolute best call at the moment, it just seems a little odd to have an argument not be valid for the majority of implementations (and probably most others that may be added - please correct me if this seems off). If you feel strongly about the current approach, I am okay with it.

Reply:
An additional interface/implementation feels like overkill. As for the dependency injection, we can likely solve this with documentation - saying that the lock *should* support shared reading. In this case of the `BreakpointStorageInterfaces`, we don't provide lock injection - if you're using the file based method, then we use the flock lock implementation.

I'll move the exclusive option to the constructor and out of the interface.